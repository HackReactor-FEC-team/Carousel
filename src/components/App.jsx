import React from 'react';
import './main.css';
// import axios from 'axios';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <h1>Recommended products</h1>

      </div>
    );
  }
}

export default App;
