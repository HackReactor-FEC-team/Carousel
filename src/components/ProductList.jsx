/* eslint-disable react/no-array-index-key */
/* eslint-disable react/prop-types */
import React from 'react';
import Product from './Product';

const ProductList = (props) => (

  <div>
    {
    props.items.map((itemObj, i) => <Product item={itemObj} key={i} />)
    }
  </div>

);

export default ProductList;
