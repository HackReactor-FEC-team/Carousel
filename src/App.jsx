import React from 'react';
import './main.css';
import axios from 'axios';
import ProductList from './components/ProductList';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { recommendedProducts: [] };
  }

  componentDidMount() {
    axios.get('http://localhost:3001/items')
      .then((items) => {
        this.setState({ recommendedProducts: items.data });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        <h1>Recommended products!</h1>
        <ProductList items={this.state.recommendedProducts} />
      </div>
    );
  }
}

export default App;
