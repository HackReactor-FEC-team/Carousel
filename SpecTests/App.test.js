import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../src/App';
import ProductList from '../src/components/ProductList';

describe('App component', () => {
  test('renders App component', () => {
    // eslint-disable-next-line react/jsx-filename-extension
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the inner ProductList', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(ProductList).length).toEqual(1);
  });
});
