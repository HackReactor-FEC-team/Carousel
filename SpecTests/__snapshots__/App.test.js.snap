// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
<div>
  <h1>
    Recommended products!
  </h1>
  <div />
</div>
`;

exports[`App component renders App component 1`] = `
<div>
  <h1>
    Recommended products!
  </h1>
  <div />
</div>
`;
